From 1354f8412a02402c8134954f9441d66b5976404d Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Mon, 25 Feb 2019 22:12:46 +0800
Subject: [PATCH 17/23] Make custom off-mode charging screen great again

 * Move font_log.png to images dir
 * Add custom animation to work with AOSP charger
 * Make it opt-out

Change-Id: Ibc7fa15d76df393f607a8f7263d7aa7c1cb58f2b
---
 charger/Android.mk                            |   7 ++--
 charger/fonts/560dpi                          |   1 -
 charger/fonts/640dpi                          |   1 -
 charger/{fonts => images}/hdpi/font_log.png   | Bin
 charger/{fonts => images}/ldpi/font_log.png   | Bin
 charger/{fonts => images}/mdpi/font_log.png   | Bin
 charger/{fonts => images}/xhdpi/font_log.png  | Bin
 charger/{fonts => images}/xxhdpi/font_log.png | Bin
 .../{fonts => images}/xxxhdpi/font_log.png    | Bin
 charger/values/animation.txt                  |  36 ++++++++++++++++++
 config/common.mk                              |   3 +-
 11 files changed, 42 insertions(+), 6 deletions(-)
 delete mode 120000 charger/fonts/560dpi
 delete mode 120000 charger/fonts/640dpi
 rename charger/{fonts => images}/hdpi/font_log.png (100%)
 rename charger/{fonts => images}/ldpi/font_log.png (100%)
 rename charger/{fonts => images}/mdpi/font_log.png (100%)
 rename charger/{fonts => images}/xhdpi/font_log.png (100%)
 rename charger/{fonts => images}/xxhdpi/font_log.png (100%)
 rename charger/{fonts => images}/xxxhdpi/font_log.png (100%)
 create mode 100644 charger/values/animation.txt

diff --git a/charger/Android.mk b/charger/Android.mk
index c4af4317..affa15fc 100644
--- a/charger/Android.mk
+++ b/charger/Android.mk
@@ -15,11 +15,12 @@ endif
 endif
 
 include $(CLEAR_VARS)
-LOCAL_MODULE := font_log.png
-LOCAL_SRC_FILES := fonts/$(healthd_density)/font_log.png
+LOCAL_MODULE := lineage_charger_animation
+LOCAL_MODULE_STEM := animation.txt
+LOCAL_SRC_FILES := values/animation.txt
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE_CLASS := ETC
-LOCAL_MODULE_PATH := $(TARGET_ROOT_OUT)/res/images
+LOCAL_MODULE_PATH := $(TARGET_ROOT_OUT)/res/values/charger
 include $(BUILD_PREBUILT)
 
 include $(CLEAR_VARS)
diff --git a/charger/fonts/560dpi b/charger/fonts/560dpi
deleted file mode 120000
index 17ef829d..00000000
--- a/charger/fonts/560dpi
+++ /dev/null
@@ -1 +0,0 @@
-xxhdpi
\ No newline at end of file
diff --git a/charger/fonts/640dpi b/charger/fonts/640dpi
deleted file mode 120000
index af2d1fe7..00000000
--- a/charger/fonts/640dpi
+++ /dev/null
@@ -1 +0,0 @@
-xxxhdpi
\ No newline at end of file
diff --git a/charger/fonts/hdpi/font_log.png b/charger/images/hdpi/font_log.png
similarity index 100%
rename from charger/fonts/hdpi/font_log.png
rename to charger/images/hdpi/font_log.png
diff --git a/charger/fonts/ldpi/font_log.png b/charger/images/ldpi/font_log.png
similarity index 100%
rename from charger/fonts/ldpi/font_log.png
rename to charger/images/ldpi/font_log.png
diff --git a/charger/fonts/mdpi/font_log.png b/charger/images/mdpi/font_log.png
similarity index 100%
rename from charger/fonts/mdpi/font_log.png
rename to charger/images/mdpi/font_log.png
diff --git a/charger/fonts/xhdpi/font_log.png b/charger/images/xhdpi/font_log.png
similarity index 100%
rename from charger/fonts/xhdpi/font_log.png
rename to charger/images/xhdpi/font_log.png
diff --git a/charger/fonts/xxhdpi/font_log.png b/charger/images/xxhdpi/font_log.png
similarity index 100%
rename from charger/fonts/xxhdpi/font_log.png
rename to charger/images/xxhdpi/font_log.png
diff --git a/charger/fonts/xxxhdpi/font_log.png b/charger/images/xxxhdpi/font_log.png
similarity index 100%
rename from charger/fonts/xxxhdpi/font_log.png
rename to charger/images/xxxhdpi/font_log.png
diff --git a/charger/values/animation.txt b/charger/values/animation.txt
new file mode 100644
index 00000000..9a499a00
--- /dev/null
+++ b/charger/values/animation.txt
@@ -0,0 +1,36 @@
+#
+# Copyright (C) 2019 The LineageOS Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# animation: num_cycles, first_frame_repeats, animation_file
+animation: 3 2 charger/lineage_battery_scale
+
+# percent_display: text_percent_pos_x, text_percent_pos_y, color_r, color_g, color_b, color_a, font_file
+percent_display: c c 22 124 128 255 charger/font_log
+
+fail: charger/battery_fail
+
+# frame: disp_time min_level max_level
+frame: 750 0 100
+frame: 750 10 100
+frame: 750 20 100
+frame: 750 30 100
+frame: 750 40 100
+frame: 750 50 100
+frame: 750 60 100
+frame: 750 70 100
+frame: 750 80 100
+frame: 750 90 100
+frame: 750 100 100
diff --git a/config/common.mk b/config/common.mk
index 93c8a7e0..a33f420d 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -196,8 +196,9 @@ PRODUCT_PACKAGES += \
     charger_res_images
 
 # Custom off-mode charger
-ifeq ($(WITH_LINEAGE_CHARGER),true)
+ifneq ($(WITH_LINEAGE_CHARGER),false)
 PRODUCT_PACKAGES += \
+    lineage_charger_animation \
     lineage_charger_res_images \
     font_log.png
 endif
-- 
2.17.1

